frontend/hub/execution-environments/ExecutionEnvironmentForm.tsx

```diff
@@ -0,0 +1,333 @@
+import { ExecutionEnvironment } from './ExecutionEnvironment';
+import { HubPageForm } from '../HubPageForm';
+import { HubItemsResponse } from '../useHubView';
+import { useState, useCallback, useMemo } from 'react';
+
+import { patchHubRequest, putHubRequest } from './../api/request';
+import { PageFormAsyncSelect } from '../../../framework/PageForm/Inputs/PageFormAsyncSelect';
+import { useSelectRegistrySingle } from './hooks/useRegistrySelector';
+import { usePageNavigate } from '../../../framework';
+
+export function CreateExecutionEnvironment() {
+  return <ExecutionEnvironmentForm mode="add" />;
+}
+
+export function EditExecutionEnvironment() {
+  return <ExecutionEnvironmentForm mode="edit" />;
+}
+
+function ExecutionEnvironmentForm(props: { mode: 'add' | 'edit' }) {
+  const page_size = 50;
+  const { t } = useTranslation();
+  const navigate = usePageNavigate();
+  const getPageUrl = useGetPageUrl();
+  const mode = props.mode;
+  const params = useParams<{ id?: string }>();
+
+  const [tagsToInclude, setTagsToInclude] = useState<string[]>([]);
+  const [tagsToExclude, setTagsToExclude] = useState<string[]>([]);
+  const [tagsSet, setTagsSet] = useState<boolean>(false);
+
+  const registry = useGet<HubItemsResponse<Registry>>(
+    hubAPI`/_ui/v1/execution-environments/registries?page_size=${page_size.toString()}`
+  );
+
+  const eeUrl = useMemo(
+    () =>
+      mode == 'edit'
+        ? hubAPI`/v3/plugin/execution-environments/repositories/${params?.id || ''}/`
+        : '',
+    [mode, params?.id]
+  );
+
+  const executionEnvironment = useGet<ExecutionEnvironment>(eeUrl);
+
+  const singleRegistryUrl = useMemo(
+    () =>
+      mode == 'edit' && executionEnvironment.data
+        ? hubAPI`/_ui/v1/execution-environments/registries/${
+            executionEnvironment.data?.pulp?.repository?.remote?.registry || ''
+          }/`
+        : '',
+    [mode, executionEnvironment.data]
+  );
+
+  const singleRegistry = useGet<Registry>(singleRegistryUrl);
+  const isLoading = (!executionEnvironment.data || !singleRegistry.data) && mode == 'edit';
+
+  if (mode == 'edit' && !tagsSet && isLoading == false) {
+    setTagsSet(true);
+    setTagsToExclude(executionEnvironment.data?.pulp?.repository?.remote?.exclude_tags || []);
+    setTagsToInclude(executionEnvironment.data?.pulp?.repository?.remote?.include_tags || []);
+  }
+
+  const selectRegistrySingle = useSelectRegistrySingle();
+
+  const registrySelector = selectRegistrySingle.onBrowse;
+
+  const isNew = !executionEnvironment.data?.pulp?.repository;
+  const isRemote = executionEnvironment.data?.pulp?.repository
+    ? !!executionEnvironment.data?.pulp?.repository?.remote
+    : true;
+
+  const query = useCallback(async () => {
+    return Promise.resolve({
+      total: registry?.data?.meta?.count || 0,
+      values: registry?.data?.data || [],
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [registry.data]);
```

Comment:
this is suspicious .. it doesn't actually call any apis, I asume because useGet already handles that...

and `registry` is not filtering by anything, so this just returns the total count of registries and the first 50 registries

Sounds like this is old code, and both the registry `useGet` and this `query` function should be removed and are replaced by `useRegistrySelector`?

Reply:
It will be replaced in the future, I have to add this into the framework first. For now, I discovered I can remove async, but query requires  function that will fetch data and return promise, so thats why it looks suspicious. I had useGetRequest for that inside the function with await, but since Kersom and Alex told me, we should use useGet everywhere, I fetch data outside and returned its result there.